### ISSUE TITLE
bypass action on filters only bypasses current phase

### ISSUE THREAD

**Issue by cwningen** [OpenSMTPD/OpenSMTPD#1078]
Documention suggests that bypass can be used to bypass all filters in a given session/transaction but the keyword only affects current phase.

```
filter trusted phase mail-from match src <trusted> bypass
filter bad phase mail-from match !rdns disconnect "550 some message"
```

The above works, however if you try;

```
filter trusted phase connect match src <trusted> bypass
filter bad phase mail-from match !rdns disconnect "550 some message"
```

The former will work as expected if the IP is in `<trusted>`, whereas it will be disconnected in the latter despite matching the trusted filter bypass. If this is by design then the documention should reflect this.

**Comment by poolpOrg** [comment]
Yes, this is by design, the document is inaccurate if it states otherwise, I'll submit a diff upstream.

In another note, maybe a mechanism to bypass all filters from all phases should be provided, I'll think about something I can submit a diff upstream for.

**Comment by cwningen** [comment]
Ahh that's fine :) I think this is my only complaint really but it's been no bother doing this all at rcpt-to or data stages.

**Comment by law** [comment]
I believe I'm running into this as well.  I'm trying to set up a table to hold 'trusted' IPs that don't have fully-configured mailservers (with FcRDNS, etc), but the 'bypass' keyword doesn't seem to allow them to deliver still.  Happy to provide more info if desired, or I can sit tight until the diff upstream lands :-)

**Comment by biergaizi** [comment]
Another related problem is that mixing a built-in filter with `bypass` and an external `proc-exec` doesn't seem to work.

The following filter chain attempts to run `do_dkimsign` only for authenticated users, this is useful when trusted users and public servers need to submit mails to the same SMTP port - one should sign mails only for authenticated users.

    filter ignore_noauth phase data match !auth bypass
    filter do_dkimsign \
            proc-exec "filter-dkimsign -d example.com -s 42 -k /etc/mail/dkim/private.rsa.key" \
            user _dkimsign group _dkimsign

    filter dkimsign_authonly chain { ignore_noauth, do_dkimsign }

    listen on 0.0.0.0 tls   pki "example.com" auth-optional filter dkimsign_authonly
    listen on ::      tls   pki "example.com" auth-optional filter dkimsign_authonly
    listen on 0.0.0.0 smtps pki "example.com" auth-optional filter dkimsign_authonly
    listen on ::      smtps pki "example.com" auth-optional filter dkimsign_authonly
    listen on 0.0.0.0 port  submission tls-require pki "example.com" auth-optional filter dkimsign_authonly
    listen on ::      port  submission tls-require pki "example.com" auth-optional filter dkimsign_authonly

According to the source code of `filter-dkimsign`, it only hooks into the `dataline` and `commit` phases of a mail. But the `bypass` rule doesn't seem to work, and all incoming mails are still signed... A "total reject" feature would solve this problem.

Given the design limitation, it seems that right now the only solution is patching `filter-dkimsign` myself to implement a "sign only when authed" feature.
